### PR TITLE
Tdimhcsleumas/login flow

### PR DESCRIPTION
**Pull request by tdimhcsleumas**
- login is now a model which returns to the launch screen
- on login, a table view is launched as a modal to select a default team
- The side bar was modified to accept an identity as input, and passes that information along to the TeamTableViewController
- Deleted the originally SluggoTableViewController since TeamTableViewController essentially covers all of its functionality
- ~~Modified the AppIdentity to use computed values and queue up writes to the file every time a value is change. Not the most efficient but it seems to work.~~ this is getting reverted, I'm tired
- There is a button that does nothing in the TeamTableViewContainer, but it might be a good place to enter an invite code or something

### CODE REVIEW

File: Sluggo/Models/AppIdentity.swift [hunk 1]
```diff
@@ -8,11 +8,69 @@
...
+    var baseAddress: String {
+        set(newBaseAddress) {
+            _baseAddress = newBaseAddress
+            enqueueSave()
+        }
+        get {
+            return _baseAddress
+        }
+    }
+    var rememberMe: Bool {
```
Comment (wtpisaac) [inline Sluggo/Models/AppIdentity.swift:65]: This really ought not to trigger a save, as if Remember Me is deselected, then we will not enqueue any sort of save, because we have disabled saving.

Comment (wtpisaac) [reply Sluggo/Models/AppIdentity.swift:65]: or at the very least, by nature of the app design, we will not have the user entering this beyond the login screen

File: Sluggo/Models/AppIdentity.swift [hunk 2]
```diff
@@ -43,7 +107,22 @@ class AppIdentity: Codable {
     }
     
     func saveToDisk() -> Bool {
-        guard let appIdentityData = JsonLoader.encode(object: self) else {
+        
+        let savedRecord = AppIdentity()
+        
+        savedRecord._pageSize = self._pageSize
+        savedRecord._baseAddress = self._baseAddress
```
Comment (wtpisaac) [inline Sluggo/Models/AppIdentity.swift:114]: Base address may eventually be put directly on the login screen. We should decide if this would count as personally identifying information, since it could theoretically leak information about what teams the user is on.

(Or, frankly, consider nixing remember me, but that's beyond the scope of this branch of course.)

File: Sluggo/Models/Codables/Team.swift [hunk 3]
```diff
@@ -8,13 +8,21 @@
...
     var object_uuid: UUID
     var ticket_head: Int
     var created: Date
     var activated: Date?
     var deactivated: Date?
-   
+    
+    static func == (lhs: TeamRecord, rhs: TeamRecord) -> Bool {
+        return lhs.id == rhs.id &&
+            lhs.object_uuid == rhs.object_uuid &&
```
Comment (wtpisaac) [inline Sluggo/Models/Codables/Team.swift:22]: if the UUID is indeed unique, could we not save time by simply comparing the UUID? or is there a specific intended purpose of this equality check?

File: Sluggo/View Controllers/TeamTableViewController.swift [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        refreshControl = UIRefreshControl()
+        refreshControl?.addTarget(self, action: #selector(handleRefreshAction), for: .valueChanged)
+    }
+    
+    @objc func handleRefreshAction() {
+        teams = []
+        self.loadData(page: 1)
+    }
+    
+    private func preselectRow() {
```
Comment (wtpisaac) [inline Sluggo/View Controllers/TeamTableViewController.swift:32]: Encountered a fatal error here: In my testing this crashes if there are no teams currently on the instance. This is OK if we are going to address this in a future commit, but ideally we could add some error to prevent this code path from executing.

File: Sluggo/View Controllers/RootViewController.swift [hunk 5]
```diff
@@ -56,4 +58,14 @@ class RootViewController: UIViewController {
     @IBSegueAction func createTicket(_ coder: NSCoder) -> TicketListController? {
         return TicketListController(coder: coder, identity: identity)
     }
+    
+    @IBAction func receivedGesture() {
+        print("swiped right")
```
Comment (wtpisaac) [inline Sluggo/View Controllers/RootViewController.swift:63]: is this placeholder functionality? is this not intended to expand/collapse the sidebar?

File: Sluggo/View Controllers/TeamSelectorContainerViewController.swift [hunk 6]
```diff
@@ -0,0 +1,59 @@
+//
```
Comment (wtpisaac) [inline Sluggo/View Controllers/TeamSelectorContainerViewController.swift:1]: Not sure what the intent of this container view is. It appears to wholly embed a navigation controller, in which case we could just subclass UINavigationController in such a way to enable injecting of identity. Let me know if I'm missing something here.

Comment (wtpisaac) [reply Sluggo/View Controllers/TeamSelectorContainerViewController.swift:1]: (That said, this should be fine, it works for what we're doing. Navigation controller subclassing would be more effort, not necessarily needed)

**Review by wtpisaac — APPROVED** [review]
Looks good, although please review the comments I've left. I think this is good enough, it does pass tests under ideal circumstances. I did encounter a crash with an edge case though that may be good to resolve, but this could also be deferred until we work on invitation logic.

File: Sluggo/Models/AppIdentity.swift (hunk 1 above)
Comment (tdimhcsleumas) [reply Sluggo/Models/AppIdentity.swift:65]: no! I think we should persist the url differently because we could have different users login to the same device. Remember me to me sounds more like saving the current user, not the url.

File: Sluggo/Models/AppIdentity.swift (hunk 2 above)
Comment (tdimhcsleumas) [reply Sluggo/Models/AppIdentity.swift:114]: This is necessary for app state. Both of these should be saved no matter what and then changed where necessary

File: Sluggo/Models/Codables/Team.swift (hunk 3 above)
Comment (tdimhcsleumas) [reply Sluggo/Models/Codables/Team.swift:22]: there is no specific purpose I'm just dumb

File: Sluggo/View Controllers/TeamTableViewController.swift (hunk 4 above)
Comment (tdimhcsleumas) [reply Sluggo/View Controllers/TeamTableViewController.swift:32]: I didn't test for that, will resolve

File: Sluggo/View Controllers/RootViewController.swift (hunk 5 above)
Comment (tdimhcsleumas) [reply Sluggo/View Controllers/RootViewController.swift:63]: I didn't because I was under the assumption that you were eventually going to handle this. I can take care of it.

File: Sluggo/View Controllers/TeamSelectorContainerViewController.swift (hunk 6 above)
Comment (tdimhcsleumas) [reply Sluggo/View Controllers/TeamSelectorContainerViewController.swift:1]: I'm using it to launch itself modally using  constructor injection. I can revisit this to see if there's a better way.

File: Sluggo/View Controllers/RootViewController.swift (hunk 5 above)
Comment (wtpisaac) [reply Sluggo/View Controllers/RootViewController.swift:63]: I will handle on my branch
